### ISSUE TITLE
[BUG]: migrate-locations command - Subscription ID error

### ISSUE THREAD

**Issue by heydil** [databrickslabs/ucx#1632]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Encountering an error when attempting to run the migrate-locations command. The command throws the following error message:

ValueError: Please enter subscription id to scan storage accounts in.

I've reviewed the documentation on GitHub and included the subscription ID flag while running the command. However, this doesn't seem to resolve the issue.

![image](https://github.com/databrickslabs/ucx/assets/76246320/4b3f1f15-8e8d-4002-ba8a-5df9c00620f6)

Generated the azure_storage_account_info.csv file using principal-prefix-access command, and run both migrate-credentials and validate-external-locations commands successfully.

### Expected Behavior

According to the documentation, the migrate-locations command should not prompt for a subscription ID. It should be able to identify the relevant subscription itself. Additionally, should successfully create the external location in the Databricks workspace.

### Steps To Reproduce

1. Configured Databricks profile: Added the subscription ID and set auth_type=azure_cli.
2. Installed and configured Azure CLI via az-login.
3. Verified the external location exists in the dashboard/hive_metastore.ucx.external_locations table.
![image](https://github.com/databrickslabs/ucx/assets/76246320/80084304-fbdf-4c56-875a-035497d51ec0)
5. Generated the azure_storage_account_info.csv file using the principal-prefix-access command.
6. Successfully executed migrate-credentials command.
![image](https://github.com/databrickslabs/ucx/assets/76246320/304336ba-8227-4b7b-a332-c67048fa415e)
8. Successfully ran databricks labs ucx validate-external-locations, which created the TF file.
![image](https://github.com/databrickslabs/ucx/assets/76246320/18f65a57-499c-4daf-9476-a0d56aef439e)
9. Used the same Databricks configuration file (~/.databrickscfg) throughout the entire process.

### Cloud

Azure

### Operating System

Windows

### Version

latest via Databricks CLI

### Relevant log output

```shell
ERROR [src/databricks/labs/ucx.migrate-locations] ValueError: Please enter subscription id to scan storage accounts in.
```

**Comment by gregorymiguel** [comment]
Hi @heydil 
They are fixing this issue here #1619.

**Comment by nkvuong** [comment]
closed by #1619
